sql/parser/builtins.go

```diff
@@ -761,3 +774,12 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+func round(x float64, n int64) (Datum, error) {
+	if n < 0 || n > 50 {
```

Comment:
By "allowed", I mean that if the number of digits is too high we can just return the original number immediately instead of returning an error. We don't need to actually call `strconv.FormatFloat` if `ndigits > 308`.